server/service/appconfig.go

```diff
@@ -76,6 +76,14 @@ func getAppConfigEndpoint(ctx context.Context, request interface{}, svc fleet.Se
 		hostExpirySettings = config.HostExpirySettings
 		agentOptions = config.AgentOptions
 	}
+
+	transparencyURL := fleet.DefaultTransparencyURL
+	// Fleet Premium license is required for custom transparency url
+	if license.Tier == "premium" && config.FleetDesktop.TransparencyURL != "" {
+		transparencyURL = config.FleetDesktop.TransparencyURL
+	}
+	fleetDesktop := fleet.FleetDesktopSettings{TransparencyURL: transparencyURL}
```

Comment:
@lucasmrod, I agree it gets confusing. And the location of the logic has changed some while the PR has been open. 

In its current form, it is arranged as follows:
- endpoint responses are where we substitute the default URL in the case of an empty config value or a non-premium user
- `svc.AppConfig` returns the URL as stored in the db
- `svc.ModifyAppConfig` enforces the rule that stored value for `transparency_url` must be "" for non-premium users

In order to make the default stored value "" to give us the future flexibility you [suggested](https://github.com/fleetdm/fleet/pull/6151#discussion_r892902544), the logic was split because `svc.ModifyAppConfig` calls `svc.AppConfig` as the beginning and end of the modify [method](https://github.com/fleetdm/fleet/pull/6151/files/a9bd6c711f8c2602969e86fb6cef2762fab1f4a4#diff-ff669b9f96ea80679f4651e9cf45ded57d5cd939d1e4e24977eb72d37d71e8bcR186). Definitely open to ideas on how to improve it.

Reply:
As it is, `svc.ModifyAppConfig` is already quite complicated so I'm not sure the best way to approach this piece.

Reply:
> As it is, svc.ModifyAppConfig is already quite complicated so I'm not sure the best way to approach this piece.

Agree on this one. To improve this we may need a small refactor.